src/Servers/Kestrel/Core/src/Internal/Http2/Http2Connection.cs

```diff
@@ -1400,6 +1436,71 @@ private enum PseudoHeaderFields
             Unknown = 0x40000000
         }
 
+        private struct Http2StreamStack
```

Comment:
Nit: Http2Connection is already kinda large. Can we move this into its own file and link to https://github.com/dotnet/runtime/blob/843d66b238817c7bdbe42556bb17ab1e9f10973a/src/libraries/System.IO.Pipelines/src/System/IO/Pipelines/BufferSegmentStack.cs for reference?